contracts/AccessControl.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: 2023 Lido <info@lido.fi>
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+/**
+ * @title AccessControl
+ *
+ * @dev Provides access control by creating agent and manager roles.
+ * The agent is set during contract deployment and cannot be changed. The
+ * manager can be set by the agent.
+ */
+contract AccessControl {
+    address public immutable agent;
+    address public manager;
+
+    error InvalidAgentAddress();
+    error InvalidManagerAddress();
+    error NotAgentOrManager();
+    error NotAgent();
+
+    /**
+     * @dev Initializes the contract setting the deployer as the initial agent.
+     */
+    constructor(address agent_) {
+        if (agent_ == address(0)) revert InvalidAgentAddress();
+        agent = agent_;
+    }
+
+    /**
+     * @dev Sets the manager address.
+     * @param manager_ The address of the new manager.
+     */
+    function setManager(address manager_) external onlyAgent {
+        if (manager_ == address(0)) revert InvalidManagerAddress();
```

Comment:
let's have an ability to dismiss manager still

Reply:
Does changing manager to agent address work? Allowing to set zero address sounds a little bit scary to me.